src/fe-small-advertisements/src/components/App.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { changeNameActionCreator } from '../actions';
+import { RootState } from '../reducers';
+
+function App() {
+  const { name } = useSelector((state: RootState) => state.user);
+  const dispatch = useDispatch();
+  const [newName, setNewName] = useState('');
+
+  const handleNewName = (e: ChangeEvent<HTMLInputElement>): void => {
```

Comment:
```suggestion
  const setNewName = (e: ChangeEvent<HTMLInputElement>): void => {
```

You basically setting a new name.